### DISCUSSION TITLE
How do you specify a type as required?

### DISCUSSION THREAD

**Discussion by miguderp** [joe-bell/cva#50] [Q&A] [answered]
Hey 👋🏻 

Using `VariantProps` we can automatically generate types but they all seem to be optional.
Is it possible to make some of them required?

Thanks!

**Comment by joe-bell** [comment, marked as answer]
It's not something I plan on adding as a feature, but you can achieve it with TypeScript's built in utility types!

Here's a very rough example

```ts
import { cva, type VariantProps } from "cva";

const buttonVariants = cva("buttonStyles…", {
  variants: {
    optional: {
      a: "",
      b: "",
    },
    required: {
      a: "",
      b: "",
    },
  },
});

type ButtonVariantProps = VariantProps<typeof buttonVariants>;

export interface ButtonProps
  extends Omit<ButtonVariantProps, "required">,
    Required<Pick<ButtonVariantProps, "required">> {}

export const button = (props: ButtonProps) => buttonVariants(props);
```

<img width="825" alt="CleanShot 2022-10-26 at 10 46 58@2x" src="https://user-images.githubusercontent.com/7349341/197902524-25e6e6e5-ab29-42bc-819a-eab7d33f2cfd.png">

**Reply by joe-bell** [reply]
Added to the README too 

https://github.com/joe-bell/cva#required-variants
